library/src/main/java/com/datatorrent/lib/db/cache/CacheManager.java

```diff
@@ -56,27 +61,55 @@
 public class CacheManager implements Closeable
 {
   @NotNull
-  protected Primary primary;
+  protected transient Primary primary;
   @NotNull
-  protected Backup backup;
+  protected transient Backup backup;
   protected String refreshTime;
   private transient Timer refresher;
 
+  private transient Attribute.AttributeMap attributeMap = new Attribute.AttributeMap.DefaultAttributeMap();
+  private boolean readOnlyAttribute = false;
+  private int numInitCachedLines = -1;
+
   public CacheManager()
   {
     this.primary = new CacheStore();
   }
 
   public void initialize() throws IOException
   {
+    Class[] parameters = new Class[1];
+    parameters[0] = CacheContext.class;
+
+    if (primary instanceof Component) {
+      try {
+        primary.getClass().getMethod("setup", parameters);
+        attributeMap.put(CacheContext.READ_ONLY_ATTR, readOnlyAttribute);
+        attributeMap.put(CacheContext.NUM_INIT_CACHED_LINES_ATTR, numInitCachedLines);
+        ((Component)primary).setup(new CacheContext(attributeMap));
```

Comment:
How about reusing the CacheContext instance between the primary and backup initialization.